src/Common/src/System/Net/Security/SslClientAuthenticationOptionsExtensions.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Net.Security
+{
+    internal static class SslClientAuthenticationOptionsExtensions
+    {
+        public static SslClientAuthenticationOptions ShallowClone(this SslClientAuthenticationOptions options) =>
+            new SslClientAuthenticationOptions()
+            {
+                AllowRenegotiation = options.AllowRenegotiation,
+                ApplicationProtocols = options.ApplicationProtocols,
```

Comment:
ApplicationProtocols doesn't make sense to configure externally, ManagedHandler currently only supports one value (HTTP/1.1). Even when you add HTTP/2 support there's no reason for the developer have open-ended configuration here. Unless you want to use this as your on/off switch for HTTP/2? ApplicationProtocols does not seem like the most user friendly API for controlling that.

Reply:
If that is the case then you will need to override "allow renegotiation" as Http/2 doesn't allow it

Reply:
Are you at least going to check the ApplicationProtocols before or after the handshake and fail if they contain unsupported protocols? E.g. anything other than HTTP/1.1 for now. It would get pretty messy if the dev specified HTTP/2 and the server started sending you gibberish.